sdk/eventhub/azure-eventhubs/azure/eventhub/_consumer_client.py

```diff
@@ -0,0 +1,182 @@
+         partition is closed. The callback takes two parameters: `partition_context` which contains partition
+         information and `reason` for the close. Please define the callback like `on_error(partition_context, reason)`.
+         Please refer to `azure.eventhub.CloseReason` for different closing reason.
+        :rtype: None
+
+        .. admonition:: Example:
+
+            .. literalinclude:: ../samples/sync_samples/sample_code_eventhub.py
+                :start-after: [START eventhub_consumer_client_receive_sync]
+                :end-before: [END eventhub_consumer_client_receive_sync]
+                :language: python
+                :dedent: 4
+                :caption: Receive events from the EventHub.
+        """
+        partition_id = kwargs.get("partition_id")
+
+        with self._lock:
+            error = None
+            if (consumer_group, '-1') in self._event_processors:
+                error = ValueError("This consumer client is already receiving events from all partitions for"
+                                   " consumer group {}. "
+                                   "Cannot receive from any other partitions again.".format(consumer_group))
+            elif partition_id is None and any(x[0] == consumer_group for x in self._event_processors):
+                error = ValueError("This consumer client is already receiving events for consumer group {}. "
+                                   "Cannot receive from all partitions again.".format(consumer_group))
+            elif (consumer_group, partition_id) in self._event_processors:
+                error = ValueError("This consumer is already receiving events from partition {} for consumer group {}. "
+                                   "Cannot receive from it again.".format(partition_id, consumer_group))
+            if error:
+                log.warning(error)
+                raise error
+
+            event_processor = EventProcessor(
+                self, consumer_group, on_event,
+                partition_manager=self._partition_manager,
+                polling_interval=self._load_balancing_interval,
+                **kwargs
+            )
+            self._event_processors[(consumer_group, partition_id or "-1")] = event_processor
+
+        event_processor.start()
+
+    def close(self):
+        # type: () -> None
+        """Stop retrieving events from event hubs and close the underlying AMQP connection and links.
+
+        :rtype: None
+
+        .. admonition:: Example:
+
+            .. literalinclude:: ../samples/sync_samples/sample_code_eventhub.py
+                :start-after: [START eventhub_consumer_client_close_sync]
+                :end-before: [END eventhub_consumer_client_close_sync]
+                :language: python
+                :dedent: 4
+                :caption: Close down the client.
+
+        """
+        with self._lock:
```

Comment:
I don't see where `self._lock` is initialized?

Reply:
it's initialized in the` __init__` of EventHubClient.